HeterogeneousCore/CUDACore/README.md

```diff
@@ -341,6 +342,65 @@ void ProducerInputOutputCUDA::produce(edm::StreamID streamID, edm::Event& iEvent
 }
 ```
 
+### Analyzer with CUDA input
+
+Analyzer with CudA input is similar to [producer with CUDA input](#producer-with-cuda-input)
+
+```cpp
+class AnalyzerInputCUDA: public edm::global::EDAnalyzer<> {
+public:
+  ...
+  void analyzer(edm::Event const& iEvent, edm::EventSetup const& iSetup) override;
+  ...
+private:
+  ...
+  AnalyzerInputGPUAlgo gpuAlgo_;
+  edm::EDGetTokenT<CUDAProduct<InputData>> inputToken_;
+  edm::EDGetTokenT<CUDAProduct<OtherInputData>> otherInputToken_;
+};
+...
+void AnalyzerInputCUDA::analyze(edm::Event const& iEvent, edm::EventSetup& iSetup) {
+
+  // Currently there is no specific support for CUDA-analyzers in the
```

Comment:
@makortel can you elaborate on this point ?
If the instance of the `AnalyzerInputCUDA` depends on a product, won't the framework always schedule the relevant `EDProducer` before the analyzer (assuming unscheduled configuration, or an equivalently schedules one) ?

Reply:
What I mean is that there is no `SwitchAnalyzer`. The framework indeed handles the production of inputs properly. But if the configuration of the analyzer is run on a machine without a GPU, the framework will still run the analyzer but not the producers, and this situation must be dealt with explicitly in the analyzer (if the author wants the job configuration to technically work also without a GPU).

Reply:
Mhm, but if the analyzer depends on a collection and there is no producer for it, won't the framework complain ?

Reply:
The GPU producer is there (in a Task). And what actually happens is that if a configuration with the analyzer is being run without a GPU, the analyzer will trigger the execution of the GPU producer (*) leading to a crash. So it follows that a GPU analyzer is as GPU-specific as GPU producer, and for a "portable" configuration we'd need support from the framework (so the question becomes, do we need "portable" configuration for analyzers?). I'll anyway clarify the current situation in the example code and README.

(*) with SwitchProducer the switch sets up the prefetching such that the GPU producer(s) are not run without a GPU

Reply:
OK, what I meant is that the check is not supposed to fail - either it works, or the job will have crashed before reaching it.